### ISSUE TITLE
About Model and Data Details

### ISSUE THREAD

**Issue by JunjieLl** [stepfun-ai/Step-Audio2#7]
Thank you for sharing such excellent work!

I have a few questions—if it’s convenient, I’d greatly appreciate your clarification:

1. **What is the total size of the model?**  
   Could you also share the size breakdown of each component, such as the:
   - Backbone LLM  
   - Audio Encoder  
   - Flow-Matching Module  
   - HiFi-GAN Vocoder  

2. **Audio Encoder Architecture & Training:**  
   I noticed that the Audio Encoder is trained in a multi-task manner. Could you elaborate on its architecture—how is it designed to support multiple tasks?  
   - For example, is it trained with different heads simultaneously?  
   - Also, how large is the training dataset?

3. **Regarding the Pretraining Stage:**

   a. **Modality Alignment with ASR Data:**  
      - Does this mean the input is speech (to be recognized), the prompt is text or audio, and the output is text tokens?

   b. **After Vocabulary Expansion:**
   
      - **I. TTS Task:**  
        - Is the input just text?  
        - Is the prompt in text or audio form?  
        - Is the output in audio tokens?
   
      - **II. Utterance-Level Text-Speech Interleaved Continuation Data:**  
        - Does this mean the loss is computed over the entire sequence?

      - **III. Clarification Requests:**  
        - Could you please clarify what *speech-to-text continuation* refers to?  
        - What does *utterance-level text-speech interleaved conversation* mean?

Thank you again for your time!

**Comment by JunjieLl** [comment]
There is another question: What type of text data is generally used for training, such as instruction text data?

**Comment by petronny** [comment]
> 1. **What is the total size of the model?**
* Backbone LLM: Smaller than Step-Audio (130B)
* Audio Encoder: Same to the encoder in Whisper 
* Flow-Matching Module: ~160M
* HiFi-GAN Vocoder: Same to the vocoder in CosyVoice 2

> 2. **Audio Encoder Architecture & Training:**
>        I noticed that the Audio Encoder is trained in a multi-task manner. Could you elaborate on its architecture—how is it designed to support multiple tasks?

It's similar to the training of Whisper, but we integrate more tasks than just transcription. Each task has a unique task label and all output is in text form.

> 3. **Regarding the Pretraining Stage:**

First of all, there is no prompt or role (eg. system/user/assitant) during pretraining.

>    a. **Modality Alignment with ASR Data:**
>   * Does this mean the input is speech (to be recognized), the prompt is text or audio, and the output is text tokens?

The input is speech, no prompt, the output are text tokens.

>    b. **After Vocabulary Expansion:**
>    * **I. TTS Task:**
>          * Is the input just text?
>          * Is the prompt in text or audio form?
>          * Is the output in audio tokens?

The input is text, no prompt, the output are audio tokens.

>    * **II. Utterance-Level Text-Speech Interleaved Continuation Data:**
>         * Does this mean the loss is computed over the entire sequence?

It is computed over the entire sequence, but only for the discrete tokens. There is no loss for latent features.

>    * **III. Clarification Requests:**
>      * Could you please clarify what _speech-to-text continuation_ refers to?

It requires the model to perform text continuation based on a given speech.
>    * What does _utterance-level text-speech interleaved conversation_ mean?

It's similar to Utterance-Level Text-Speech Interleaved Continuation, but the entire sequence is also a conversation.

> What type of text data is generally used for training, such as instruction text data?

Same to the text data for textual LLM pretraining/SFT. Again, the data for pretraining has no prompt or role.

**Comment by xxchauncey** [comment]
Hi there, do you plan to make model open-source?

**Comment by JunjieLl** [comment]
Did you use unimodal continuation during pre-training, or were all the training data text-audio pairs?

**Comment by petronny** [comment]
> Did you use unimodal continuation during pre-training, or were all the training data text-audio pairs?

Yes, we have pure speech continuation and pure text continuation.

**Comment by zhanglq95** [comment]
> First of all, there is no prompt or role (eg. system/user/assitant) during pretraining.
Hi，I found the ASR example in the example_base.py gave some prompts before audio : '请记录下你所听到的语音内容。'. Is this prompt used in pretraining?  Thanks for your help！ @petronny

**Comment by petronny** [comment]
> Hi，I found the ASR example in the example_base.py gave some prompts before audio : '请记录下你所听到的语音内容。'. Is this prompt used in pretraining? 

Yes, this is used in pretraining. I may have some misunderstanding before. I've updated my previous comment.
